### DISCUSSION TITLE
[Design/General] Urn map voting

### DISCUSSION THREAD

**Discussion by moxian** [ParadiseSS13/Paradise#17869] [Documents]
# Urn map voting

TL;DR: just read the "What this proposal IS" section, ignore the rest.

# Preface
There is an idea circling around about implementing weighted map voting for Paradise. Some people like it, others not so much. But importantly, it turns out that different people think of different things when speaking of "weighted map voting"! This contributes to some amount of confusion and uncertainty regarding the idea.

This proposal describes one of those "weighted map voting" implementations (renamed to "Urn", in order to differentiate from the others and lessen confusion), and hopes to convince the skepticts, that it is a good system.

# Goals
- Promote map diversity; have less of BoxStation (NSS Cyberiad) and more of others
- ..while being democratic and respecting the will of the people

Before i describe the "good" weighted map voting system, I need to first describe the "bad" one, and mention that it is NOT the one I'm proposing:

# What this proposal is NOT

The naive "weighted map voting" goes like this:

1) Assign different "vote weights" to different maps. Say, box gets weighted at 1x point per vote, Delta gets 1.5x, and Meta gets 2x points per vote
2) After the map vote is concluded, the amounts of votes get multiplied by the abovementioned multipliers.
For example, if 10 people voted for Box, 7 voted for Delta, and 6 for delta, then that would result in 10 "points" for Box, 10.5 for Delta, and 12 for Meta
3) The map with the most points (rather than votes) gets chosen as the next map.

This idea - while being the most obvious - has an obvious downside: How do we choose the weights, and who has the authority over those?
If the multipliers are too low, then we will still not be getting anything other than Box; and if they are too high - why have this farce of a vote in the first place?

Anyway, this is not what I'm proposing here.

# What this proposal IS

It goes like this

1) At the end of the round, poll the people for their desired next map, as per usual.
2) Collect the votes, then randomly select one of the options *weighted* by the amount of votes that option got.

To illustrate, let's say 50 people participated in the poll, 40 of those voted for Box, 6 voted for Delta, and 4 - for Meta.
That would result in Box being chosen with 80% probability, Delta - with 12% and Meta with 8%

Think of it as this: Imagine votes are paper ballots with the desired map written on them. We collect all of those into a one big urn (hence the name), shuffle them thoroughly, then draw one lucky winner. 
(Math folks might recognize the https://en.wikipedia.org/wiki/Urn_problem in this)

# Why is this good for the game

The current system in which a simple majority wins, is what is referred to as "First past the post" system. There's been loads of research papers, angry blog posts, and meme youtube videos made on why it's bad. Wikipedia has an amount of issues listed: https://en.wikipedia.org/wiki/First-past-the-post_voting

I want to discuss just two specific problem with it, which are applicable to our spess game:

## Variety paradox
Imagine for a moment a 100% hypothetical situation, in which we added literally 500 more maps, which are all the exact copy of Box. Since there's no reason to prefer Box#48 over Box#241, the end-round votes end up spread between all of them, each getting 0-1 votes. So every round ends up being Delta/Meta despite the simple majority wanting Box.
Now, to come back from the hypothetical to the reality: There is a (small, but) vocal group of players who would vote for "anything but box!" on the grounds that Box is old and boring, and they are just tired of it. They *want* new maps. But, paradoxically, adding more maps to the rotation would make them get *more Box*, not less, since those "anti-box" votes end up equally spread between Delta/Meta/Shepard!

This problem does not happen under the proposal: If 20% of people vote for non-Box, then 20% of the time you'll get non-box. Likewise, if 80% of people vote for one of the 500 Box clones, then 80% of the time we'll play on one of the 500 Box clones.

## Vote apathy
In the current system, if you want to play Delta, but know that many more people vote Box than Delta, then you you have little reason to vote yourself, as your vote doesn't change anything. Which makes people not vote alternative maps in the first place. Which creates a vicious cycle.
It's problematic because voting results do not even represent player *preferences* anymore, muddening any sort of analysis of what maps *do* even like.

With the proposal, however, every single vote matters, and does so exactly as much as any other one, which reduces the apathy (and people should only not vote if they don't care about the map choice, not because it doesn't matter)

## Democratic representation
I'm not sure how to put it other than: OBVIOUSLY, this proposed system is democratic, since every vote matters exactly as much as every other. So the maps selected in this process are representative of the will of the people.
I hope it is obvious... Let me know if not!

And finally, it's just an elegant system, if I say so myself.

# Problems and alternatives
I envision three potential pieces of criticism. They are somewhat strawman'y, so please take with a grain of salt, and let me know of more real ones if those exist.

## Strawman 1: But I like Box!
If all people liked Box more than other maps, then we'd have Box 100% of the time, and wouldn't need a map rotation in the first place. However, demonstrably, that is not the case, and there exist people who play on this server who prefer other maps over Box.
Yes, this proposal would select alternative maps more often than presently, but that is *fair*, as people liking other maps do exist.

## Strawman 2: What if someone just trolls the vote
Imagine there's a BadMap in a rotation which is absolutely terrible and hated by all (some would say we have that, and it's Meta, but I digress). Imagine, map vote rolls in, 99 people vote for normal maps, and one tider votes for BadMap. And BadMap ends up being randomly chosen with 1% chance. Is that fair to subject the 99 people to the fate of playing BadMap?

Well, I say yes, it is fair!
First of all, if it's so terrible, why is it even in rotation? Either improve it and make it better, or remove and bury it.
Second of all, it's 1% chance. If someone were to deliberatly troll that way, they'd need to spend and unreasonable amount of time on this just for a prank. 
And third of all, there are worse pranks to do. And if we're being charitable, that one person might genuinely love BadMap, and really want to play it. Why is your opinion on the matter more important than their?

## Non quite strawman: Can it be deterministic?
RNG is somewhat hard to trust. It's hard to trust that it's fair even when it is. See: gambler's paradox. Also see: X-Com 95% memes. It's also the reason why it's not used in real-world voting.
It would be nice to have the same sort of system (where each vote matters), but without rolling dice.

One such implementation would be to remember the tallies for the non-chosen options, and add them to the voting next round. And the next one. And all the following until they *do* get chosen.
This does effectively achieve the same goal, and with very similar methods. The math checks out in theory.
My problem with this is threefold:
1) it's somewhat hard to implement in code (doable of course, but...)
2) the people voting for a map might not even be there when said map gets chosen. In fact it is possible for *nobody* in a round to vote for a specific map, yet it still get chosen simply due to the inertia of past votes.
3) the theoretical math only really works if the number of voters is constant across rounds.
To illustrate both last points at once:
If we have a highpop 100-player round in which 60% vote Box and 40% vote Delta (and Box gets chosen), followed by a lowpop 60-player one in which 60% (36 people) vote Box again, and 40% (24 people) vote Meta, then *DELTA* gets selected (due to 40 Delta votes being carried over from previous round). 

These problems *probably* can be addressed, but the solution would need to be rather sophisticated (let me know if you have an elegant one!). And if it's sufficiently sophisticated, then I am not likely to trust that it is "fair", defeating the whole point.
And there's no reason for BYOND to cheat on the dice rolls anyway..

# Unresolved questions
None? If you have any - let me know in the comments, ~~like, share and subscribe~~
